### PR TITLE
Digital post (NgDP)

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/TVIST1-436

Implements [NgDP (sf1601)](https://digitaliseringskataloget.dk/integration/sf1601) for sending digital post.

Screenshot 2023-03-02 at 16 19 47

### CODE REVIEW

File: src/Service/SF1601/MeMoHelper.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        /*
+        foreach ($digitalPost->getAttachments() as $attachment) {
+            $document = $attachment->getDocument();
+
+            $additionalDocument = (new AdditionalDocument())
+                ->setLabel($document->getDocumentName())
+                ->setFile([$this->createFile($document)]);
+            $body->addToAdditionalDocument($additionalDocument);
+        }
+//*/
```
Comment (jekuaitk) [inline src/Service/SF1601/MeMoHelper.php:94]: Is this to be used later?

File: src/Command/DigitalPostEnvelopeListCommand.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            );
+        }
+
+        return self::SUCCESS;
+    }
+
+    /**
+     * @return DigitalPostEnvelope[]
+     */
+    private function findEnvelopes(InputInterface $input): array
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply src/Command/DigitalPostEnvelopeListCommand.php:50]: I intended to use for filtering envelopes, but that never happened (yet). I'll clean up.

File: src/Service/SF1601/MeMoHelper.php (hunk 1 above)
Comment (rimi-itk) [reply src/Service/SF1601/MeMoHelper.php:94]: It's to be used now! I'll clean up.

File: src/MessageHandler/DigitalPostMessageHandler.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace App\MessageHandler;
+
+use App\Entity\DigitalPost;
+use App\Message\DigitalPostMessage;
+use App\Repository\DigitalPostRepository;
+use App\Service\SF1601\DigitalPoster;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\LoggerInterface;
+use function Safe\json_encode;
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply src/MessageHandler/DigitalPostMessageHandler.php:11]: They probably should, but in this case I know that the message can be encoded as JSON.

**Review by jekuaitk — APPROVED** [review]

**Review by jekuaitk — APPROVED** [review]

**Comment by jekuaitk** [comment]
Looks good. 

We should consider what these changes to Digital Post (envelope) means for `CaseEvent`'s.

**Review by jekuaitk — APPROVED** [review]

**Pull request merged**